src/lib/menu/menu-trigger.ts

```diff
@@ -95,16 +96,34 @@ export class MatMenuTrigger implements AfterContentInit, OnDestroy {
    * @breaking-change 8.0.0
    */
   @Input('mat-menu-trigger-for')
-  get _deprecatedMatMenuTriggerFor(): MatMenuPanel {
-    return this.menu;
-  }
-
+  get _deprecatedMatMenuTriggerFor(): MatMenuPanel { return this.menu; }
   set _deprecatedMatMenuTriggerFor(v: MatMenuPanel) {
     this.menu = v;
   }
 
   /** References the menu instance that the trigger is associated with. */
-  @Input('matMenuTriggerFor') menu: MatMenuPanel;
+  @Input('matMenuTriggerFor')
+  get menu() { return this._menu; }
+  set menu(menu: MatMenuPanel) {
+    if (menu === this._menu) {
+      return;
+    }
+
+    this._menu = menu;
+    this._menuCloseSubscription.unsubscribe();
+
+    if (menu) {
+      this._menuCloseSubscription = menu.close.asObservable().subscribe(reason => {
+        this._destroyMenu();
```

Comment:
I believe that you need to add **this._menuCloseSubscription** here to  **this._cleanUpSubscriptions();**

Reply:
It's not required, because the `close` stream gets completed when the menu panel is destroyed.